src/index/shared/detect-lonely-files.ts

```diff
@@ -0,0 +1,85 @@
+import path from 'path';
+
+const extractParentDirectory = (destination: string): string | undefined => {
+  const parts = destination.split(path.sep);
+
+  if (parts.length === 1) {
+    return;
+  }
+
+  parts.pop();
+
+  return parts.join(path.sep)
+}
+
+const moveUp = (destinationPath: string) => {
+  const parts = destinationPath.split(path.sep);
+
+  return [
+    ...parts.slice(0, parts.length - 2),
+    parts[parts.length - 1]
+  ].join(path.sep)
+}
+
+export const detectLonelyFiles = (tree: Record<string, string>): Record<string, string> => {
```

Comment:
I still need a better name for this function

Reply:
The current name is ok for me 🤷‍♂

Reply:
The current name is ok for me 🤷‍♂